client/src/pages/login.js

```diff
@@ -1,12 +1,99 @@
 import React, { Component } from 'react';
+import { withRouter } from 'react-router-dom';
+import Form from 'react-bootstrap/Form';
+import Button from 'react-bootstrap/Button';
+import Api from '../services/api';
+import '../styles/login.css';
 
 
 class Login extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      email: '',
+      password: '',
+      errorMessage: '',
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+
+  // eslint-disable-next-line react/sort-comp
+  validateForm() {
+    const { email } = this.state;
+    const { password } = this.state;
+    return email.length > 0 && password.length > 7;
+  }
+
+  handleChange(event) {
+    this.setState({
+      [event.target.id]: event.target.value,
+    });
+  }
```

Comment:
Could you explain what this code is for?

Reply:
for universal change of state

Reply:
Thanks!